app/controllers/passengers_controller.rb

```diff
@@ -0,0 +1,154 @@
+class PassengersController < ApplicationController
+  def index
+    @passengers = Passenger.all
+  end
+
+  def show
+    @passenger = Passenger.find_by(id: params[:id])
+    if @passenger.nil?
+      redirect_to passengers_path
+      return
+    end
+  end
+
+  def new
+    @passenger = Passenger.new
+  end
+
+  def create
+    @passenger = Passenger.new(passenger_params) 
+    @passenger.is_active = true
+
+    if @passenger.save 
+      redirect_to passengers_path
+      flash[:success] = "Successfully added new passenger: #{view_context.link_to @passenger.name, passenger_path(@passenger.id) }"
+      return
+    else 
+      render :new, status: :bad_request
+      return
+    end
+  end
+
+  def edit
+    @passenger = Passenger.find_by(id: params[:id])
+
+    if @passenger.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def update
+    @passenger = Passenger.find_by(id: params[:id])
+
+    if @passenger.nil?
+      head :not_found
+      return
+    elsif @passenger.update(passenger_params)
+      redirect_to passenger_path(@passenger.id)
+      return
+    else 
+      render :edit, status: :bad_request 
+      return
+    end
+  end
+
+  def destroy
+    passenger = Passenger.find_by(id: params[:id])
+
+    if passenger.nil?
+      head :not_found
+      return
+    end
+
+    passenger.destroy
+
+    redirect_to passengers_path
+    return
+  end
+
+  def active
```

Comment:
There is a lot going on in this controller action. It all seems necessary controller work given the way you've designed your program. You might consider using a chained conditional to increase readability `if/elsif/elseif/.../else`